test/e2e_kubeadm/networking_test.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e_kubeadm
+
+import (
+	"net"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2elog "k8s.io/kubernetes/test/e2e/framework/log"
+
+	. "github.com/onsi/ginkgo"
+	// . "github.com/onsi/gomega"
+)
+
+var (
+	dualStack                    bool
+	podSubnetInKubeadmConfig     bool
+	serviceSubnetInKubeadmConfig bool
+)
+
+// Define container for all the test specification aimed at verifying
+// that kubeadm configures the networking as expected.
+// in case you want to skip this test use SKIP=setup-networking
+var _ = KubeadmDescribe("networking [setup-networking]", func() {
+
+	// Get an instance of the k8s test framework
+	f := framework.NewDefaultFramework("networking")
+
+	// Tests in this container are not expected to create new objects in the cluster
+	// so we are disabling the creation of a namespace in order to get a faster execution
+	f.SkipNamespaceCreation = true
+
+	// Tests are either single-stack (either IPv4 or IPv6 address)	or dual-stack
+	// We can determine this from the kubeadm config by looking at the number of
+	// IPs specified in networking.podSubnet
+	BeforeEach(func() {
+
+		// gets the ClusterConfiguration from the kubeadm kubeadm-config ConfigMap as a untyped map
+		cc := getClusterConfiguration(f.ClientSet)
+		// Extract the networking.podSubnet
+		if _, ok := cc["networking"]; ok {
+			d := cc["networking"].(map[interface{}]interface{})
+			if ps, ok := d["podSubnet"]; ok {
+				// If podSubnet is not specified, podSubnet cases will be skipped.
+				podSubnetInKubeadmConfig = true
```

Comment:
wdyt about testing against default podSubnet instead of skipping the test when the value in kubeadm is empty?

Reply:
There is a default `DefaultServicesSubnet = "10.96.0.0/12"` but I don't see a default PodSubnet in kubeadm. I suspect the controller-manager has a default pod subnet, but should we be checking default values outside of kubeadm?

Reply:
podSubnet is an open topic without a clear conclusion as you can see in this open issue https://github.com/kubernetes/kubeadm/issues/1011

I suggest to skip it, at least until kubeadm take a decision on how to treat it. Maybe adding a TODO comment referencing the issue

Reply:
ok for skipping and adding a comment
"kubeadm does not apply defaults for PodSubnet, so we are skipping this test because there are no kubeadm settings to be verified in this case"

Reply:
I added the comment.

Reply:
note: i closed https://github.com/kubernetes/kubeadm/issues/1011
given a warning about not passing a CIDR different from the default is not something we want to show and also given the CNI plugin step is outside of the kubeadm workflow.